### ISSUE TITLE
Exile Diary shows 0 chaos for every map in Heist

### ISSUE THREAD

**Issue by daxtep** [briansd9/exile-diary#81]
Tried the debug options, run as admin, reapply POESSID, relog, new filter, new client.txt.

Any idea what causes this?

**Comment by briansd9** [comment]
Hi, sorry about that, we'll get this fixed.

Can you send me your log file `%appdata%\exile-diary\log.txt`?

Also, does the map summary show the items you picked up?

**Comment by daxtep** [comment]
Items picked up are shown. 
Logs say "2020-09-21 04:04:45.690 [info] ItemPricer.js        > No rates found for 20200921"
[log.txt](https://github.com/briansd9/exile-diary/files/5253172/log.txt)

**Comment by briansd9** [comment]
Hmm, it looks like there is a connection problem between you and poe.ninja, getting item prices just hangs up repeatedly. (You can find them in the logs by searching `Got data from /api/data/`)

Do you experience any problems when visiting poe.ninja in a browser? Item categories not loading, etc.?

**Comment by daxtep** [comment]
Thanks for looking into it! 
I have no problems with poe ninja.

Could it be something with IPv6?

**Comment by briansd9** [comment]
Hmm, don't know enough about that, sorry.

Has anything about your Internet connection changed? From your log files, everything was working fine in Harvest league (entire price list retrieved in < 4 sec), but in Heist league it was taking longer than that for just one item type.

If the problem doesn't resolve itself, I've added some extra logging to trace the problem in v0.2.69.

**Comment by daxtep** [comment]
Thank you for your answer.
I just turned on my PC, tried the debug price thing but it didn't work. Every map I ran after that DID work tho!
Seems like I needed to restart after updating to 0.2.68 and now it works fine again :)

**Comment by daxtep** [comment]
I changed my character and the problem occurs again without me being able to fix it this time :(
I updated to the latest version and everything works except for Net Worth and the 0 in any map.

**Comment by briansd9** [comment]
Sorry about that, can you send me your log file again? If it's too big just the most recent lines (after updating) will do.

I suspect it's still a connection problem to poe.ninja, not much to be done directly other than keep trying... currently only one attempt to retrieve item prices is made, will fix in a future version.

If you'd like to try a manual workaround for now:
1) Restart the app (will try to get prices)
2) Watch the log file for output from `RateGetterV2.js`
3) If you see `Finished getting prices from poe.ninja` then it's working again
4) Otherwise, go back to step 1

**Comment by daxtep** [comment]
I'll do this tomorrow when I get home.

Thank you for your help and your amazing tool. Just subscribed to your patreon. One of the most important programms for PoE for me!

**Comment by daxtep** [comment]
[log.txt](https://github.com/briansd9/exile-diary/files/5283825/log.txt)

last lines: 

2020-09-25 17:47:42.767 [info] ItemFilter.js        > Loaded: {"nonunique":{},"unique":{},"gem":{},"map":{},"divcard":{},"prophecy":{},"oil":{},"fragment":{},"delve":{},"catalyst":{},"essence":{},"currency":{}}
2020-09-25 17:47:42.762 [info] main.js              > Done checking, character status is valid
2020-09-25 17:47:42.762 [info] main.js              > Starting components
2020-09-25 17:47:42.915 [info] main.js              > Checking for update
2020-09-25 17:47:43.099 [info] RateGetterV2.js      > Getting new rates in Heist for 20200925
2020-09-25 17:47:43.099 [info] RateGetterV2.js      > Getting prices for item type Currency
2020-09-25 17:47:45.395 [info] main.js              > Update for version 0.2.70 is not available (latest version: 0.2.70, downgrade is disallowed).

Doesn't show "Finished getting prices from poe.ninja" but shows no error either.

**Comment by briansd9** [comment]
Thanks for the subscription! Much appreciated :-)

I've just released v0.2.71, the main change in this one is displaying an explicit `Getting item prices from poe.ninja...` message in the bottom status bar. Three possible outcomes after this message:

1) `Finished getting item prices from poe.ninja` :heavy_check_mark: 
2) `Error getting item prices from poe.ninja, click on this message to try again` - should handle the case where it's an intermittent connection error
3) Nothing happens

If you get # 3, please send me your log file again :-p (Give it some time to complete though, takes about 30 seconds for me on a 10Mbps connection)

**Comment by daxtep** [comment]
Thank you!

I tried out the new version and observed the log file. It seems to just get stucked while getting the rates/prices. It randomly stops at any item type. Sometimtes at curency, sometimes at Oils, sometimes at UniqueAccessory. 
Then I disable the connection and it showed that there was an error while connecting.
Enabled it again and it got stuck at Oils.

Disabled firewall - no difference
Disabled Avast antivirus - no difference

[log.txt](https://github.com/briansd9/exile-diary/files/5284783/log.txt)

**Comment by daxtep** [comment]
I found the problem.. it's my internet connection. I tried with mobile tethering and it finished in 5seconds.
I don't know what causes this since everything else works perfectly fine. I use a FritzBox from Vodafone with DualStack IPv4 and IPv6 and even tried to disable the IPv6 connection in windows.
I even changed DNS server and tried again, no difference :(

**Comment by briansd9** [comment]
Hmm, it'd be very inconvenient to go through mobile just for this... 

Just released v0.2.72 with a secret option to possibly fix the issue, hope it works - please try these steps:

1) Wait for tomorrow (item prices are only retrieved once a day, so anytime after midnight rolls over should be OK)
2) Close app
3) Open your settings file `%appdata%\exile-diary\settings.json` and add `"useGzip" : false`, something like this

> {**"useGzip":false,**"accountName":"NAME","poesessid":"fe63936aefd0521e42a697ec98e850a1" ...

4) Open app, check if it works properly now

**Comment by daxtep** [comment]
You are the man! Thank you so much.

Looking forward to test this later.

**Comment by daxtep** [comment]
Unfortunately it didn't work. The files were downloaded compressed but it still didn't continue. 
I tried to do it with a VPN and it worked perfectly. It seems to be definitely my internet connection that is causing this issue. 
I changed my ISP a few weeks ago, that has to be the reason. Still can't explain why I can connect flawlessly to every single website and have no problems connecting to poe.ninja.

To further troubleshoot the issue, is there a possibility to get some kind up button for "reload poe.ninja rates/prices"?

Thank you for your help!

**Comment by briansd9** [comment]
Hmm, I see... just hung up on one item type again? Will check my code, I think there's still something I can do.

About your request, you never saw `Error getting item prices from poe.ninja, click on this message to try again` in the status bar?

**Comment by daxtep** [comment]
Yes, it takes about 5seconds per item category but then randomly hungs up at any category.
It doesn't abort, so the error message never popped up. Only when I disable my internet connection completely, then the error pops up.

Maybe it's possible to make use of a timer per category, so it tries the category it hang up on, again after 30 seconds or so.

**Comment by briansd9** [comment]
Yup, that's the solution I had in mind - just released an update. Let me know if the current settings (try again if hung up for 10 seconds, maximum of 10 tries) work for you.

Btw, if you still have `"useGzip" : false` in your settings file, it can be removed now.

**Comment by daxtep** [comment]
Thank you so much, that was fast!
I have to wait till tomorrow to try again, will report then.

Yes, I have removed the gzip line already.

**Comment by daxtep** [comment]
IT WORKED!!! Amazing man, it actually worked.
Thank you!

Here are the logs.
[log.txt](https://github.com/briansd9/exile-diary/files/5289567/log.txt)

**Comment by briansd9** [comment]
Good to hear :-) If everything's resolved, will close this issue in a while.

If you still have previous map runs with 0 chaos value, the "recheck map profit" debug tool should now be able to fill them in as well.

**Comment by daxtep** [comment]
I FIXED IT FINALLY!!!

The problem was my ISP + fritz box. The hidden option "Packet acceleration" was causing this issue. I found that out while searching for a solution for Escape from Tarkov not connecting at all. I deactivated the "Packet acceleration" in the router and now the prices load in seconds.

Thank you so much for your time. It did work with your workaround for the past months, but now I finally know what caused it.

**Comment by briansd9** [comment]
That's a very well hidden issue... good to hear it's working now.

Thanks for the bug report, it was a problem that needed fixing anyway. Cheers
